test/Sentry.Tests/Internals/ScopeStack/AsyncLocalScopeStackContainerTests.cs

```diff
@@ -0,0 +1,161 @@
+            {
+                container.Stack.Should().BeNull();
+
+                container.Stack = new[] {scope1};
+                await Task.Delay(10).ConfigureAwait(false);
+
+                container.Stack.Should().BeEquivalentTo(new[] {scope1});
+            });
+
+            var task3 = Task.Run(async () =>
+            {
+                container.Stack.Should().BeNull();
+
+                container.Stack = new[] {scope2};
+                await Task.Delay(10).ConfigureAwait(false);
+
+                container.Stack.Should().BeEquivalentTo(new[] {scope2});
+            });
+
+            await Task.WhenAll(task1, task2, task3);
+
+            container.Stack.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task Scopes_are_not_shared_between_sequential_async_executions()
+        {
+            // Arrange
+            var container = new AsyncLocalScopeStackContainer();
+
+            var scope1 = new KeyValuePair<Scope, ISentryClient>(
+                Substitute.For<Scope>(),
+                Substitute.For<ISentryClient>()
+            );
+
+            var scope2 = new KeyValuePair<Scope, ISentryClient>(
+                Substitute.For<Scope>(),
+                Substitute.For<ISentryClient>()
+            );
+
+            // Act & assert
+            await Task.Run(async () =>
+            {
+                container.Stack.Should().BeNull();
+
+                await Task.Delay(10).ConfigureAwait(false);
+
+                container.Stack.Should().BeNull();
+            });
+
+            await Task.Run(async () =>
+            {
+                container.Stack.Should().BeNull();
+
+                container.Stack = new[] {scope1};
+                await Task.Delay(10).ConfigureAwait(false);
+
+                container.Stack.Should().BeEquivalentTo(new[] {scope1});
+            });
+
+            await Task.Run(async () =>
+            {
+                container.Stack.Should().BeNull();
+
+                container.Stack = new[] {scope2};
+                await Task.Delay(10).ConfigureAwait(false);
+
+                container.Stack.Should().BeEquivalentTo(new[] {scope2});
+            });
+
+            container.Stack.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task Scopes_are_shared_between_nested_async_executions()
+        {
+            // Arrange
+            var container = new AsyncLocalScopeStackContainer();
+
+            var scope1 = new KeyValuePair<Scope, ISentryClient>(
+                Substitute.For<Scope>(),
+                Substitute.For<ISentryClient>()
+            );
+
+            var scope2 = new KeyValuePair<Scope, ISentryClient>(
+                Substitute.For<Scope>(),
+                Substitute.For<ISentryClient>()
+            );
+
+            // Act & assert
+            await Task.Run(async () =>
+            {
+                container.Stack.Should().BeNull();
+
+                await Task.Delay(10).ConfigureAwait(false);
```

Comment:
Do you need to wait for 10ms or just yield?
```suggestion
                await Task.Yield().ConfigureAwait(false);
```

Reply:
I don't know. I thought adding a delay would introduce some potential race conditions which could be nice to test. But I guess yield is ok too.